pkg/sql/pg_catalog.go

```diff
@@ -680,7 +680,21 @@ CREATE TABLE pg_catalog.pg_constraint (
 					confupdtype = fkActionNone
 					confdeltype = fkActionNone
 					confmatchtype = fkMatchTypeSimple
-					if conkey, err = colIDArrayToDatum(con.Index.ColumnIDs); err != nil {
+					columnIDs := con.Index.ColumnIDs
+					if int(con.FK.SharedPrefixLen) > len(columnIDs) {
+						return errors.Errorf(
```

Comment:
In both `master` and this backport, please replace this by `return pgerror.NewAssertionErrorf(...)` and simplify the sentence.

Reply:
Sure, incoming PR for master.

And I'll add the new commit to this PR.

Reply:
See #31925.